src/main/java/eu/decentsoftware/holograms/api/holograms/Hologram.java

```diff
@@ -633,21 +633,21 @@ public boolean show(@NonNull Player player, int pageIndex) {
             if (page != null && page.size() > 0 && canShow(player) && isInDisplayRange(player)) {
                 if (isVisible(player)) {
                     hide(player);
+                    return true; // Skip a tick before updating, should fix the previous issue with clients desyncing.
                 }
-                if (Version.after(8)) {
-                    showPageTo(player, page, pageIndex);
-                } else {
-                    // We need to run the task later on older versions as, if we don't, it causes issues with some holograms *randomly* becoming invisible.
-                    // I *think* this is from despawning and spawning the entities (with the same ID) in the same tick.
-                    S.sync(() -> showPageTo(player, page, pageIndex), 0L);
-                }
+
+                showPageTo(player, page, pageIndex);
                 return true;
             }
             return false;
         }
     }
 
     private void showPageTo(@NonNull Player player, @NonNull HologramPage page, int pageIndex) {
+        if (page.getLines().stream().anyMatch(line -> !line.canShow(player))) {
+            return;
+        }
```

Comment:
Are you sure this won't cause issues?

If my logic isn't failing me here would this return true if **any** line of a page can't be shown to a player... Meaning that even if only one of like 10 lines can't be shown, the page wouldn't be shown at all, which could break with existing setups using permissions per line.

Reply:
Fixed in https://github.com/DecentSoftware-eu/DecentHolograms/pull/205/commits/85a2c759ca59a1b3bc088f7b4d95d9b28f6cb370

Reply:
The main issue remains that if any line returns true, that it skips the whole page that way...

Reply:
> The main issue remains that if any line returns true, that it skips the whole page that way...

It would just send it later?

Reply:
> That isn't a good aproach imo, if the entire page is skipped because one line isn't "ready".

Well this would require a complete overhaul of the "show" logic, which would also break a good chunk of the API.

Reply:
> That isn't a good aproach imo, if the entire page is skipped because one line isn't "ready".

Could also implement some sort of "queueing" mechanism for the individual line but this is rather dirty imo.

Reply:
> That isn't a good aproach imo, if the entire page is skipped because one line isn't "ready".

https://github.com/DecentSoftware-eu/DecentHolograms/pull/205/commits/edb2adc3112cd6a5e861fecc3a29daa7af99f3fe